### PR TITLE
Webapp employee

### PR DESCRIPTION
**Pull request by rayolanderos**
All employee methods (add, delete, update, get all)
Employee dashboard layout

Missing:
Input validations and cleaning (particularily for phone number formatting)

### CODE REVIEW

**Comment by rayolanderos** [comment]
Added Order and User methods into this pull request. 
Order:
- Get all
- Get single
- Add order

User:
- Get single
- Add User
- Update User


There is also a screen in the webapp that displays all orders with details for better control

File: webapp/controllers/employees.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            "employeePhone" : employee_phone,
+            "employeeCarrier" : employee_carrier,
+            "employeeBartending" : employee_bartending
+        }
+
+        if(action == "addEmployee"):
+            api_uri = self.uri_for('add-employee', _full=True)
+        elif (action == "updateEmployee"):
+            api_uri = self.uri_for('update-employee', _full=True)
+            logging.info("++++++++++++ updating employee_data +++++++++++++++");
```
Comment (gleono) [inline webapp/controllers/employees.py]: Do you wanna keep all the logging?

File: webapp/models/order_beer.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import logging
+
+from google.appengine.ext import ndb
+
+class OrderBeer(ndb.Model):
+    
+    order_id = ndb.IntegerProperty()
+    beer_id = ndb.IntegerProperty()
```
Comment (gleono) [inline webapp/models/order_beer.py:8]: I think these 2 should be `ndb.KeyProperty()` to actually have references of the objects

File: webapp/services/add_order.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+			    reward_id = order_reward_id, 
+			    discount = order_discount, 
+			    invoice_number = order_invoice_number
+			)
+			order_key = order.put()
+			order_id = str(order_key.id())
+
+			for order_beer in order_beers:
+				order_beer = OrderBeer(
+					order_id = int( order_id ),
```
Comment (gleono) [inline webapp/services/add_order.py]: You would use the keys directly here, instead of integers.

File: webapp/services/add_user.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+			)
+			user_key = user.put()
+			user_id = str(user_key.id())
+
+			res = { "msg" : "User successfully added", "success": True, "user_id" : user_id }
+			self.response.out.write(json.dumps(res))
+
+		else:
+			res = { "msg" : "That user already exists in the system or something went wrong. Please try again.", "success": False }
+			self.response.out.write(json.dumps(res))
```
Comment (gleono) [inline webapp/services/add_user.py:39]: a post request normally doesn't write a response. Is standard to use status code instead.

File: webapp/services/get_user.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+class GetUser(webapp2.RequestHandler):
+
+    def get(self):
+
+		user_id = int( self.request.get('user_id', 0) )
+
+		user = User.get_by_id(user_id)
+
+		if user != None:
+			user_creation_date = user.creation_date
```
Comment (gleono) [inline webapp/services/get_user.py:17]: You can do it in one line, no need for extra variables. :)
`user_creation_date = user.creation_date.strftime('%m/%d/%Y')`

File: webapp/services/update_employee.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		if employee != None:
+			employee.name = employee_name
+			employee.phone = employee_phone
+			employee.carrier = employee_carrier
+			employee.bartending = employee_bartending
+
+			employee.put()
+
+			res = { "msg" : "Employee successfully updated", "success": True, "employee_id" : employee_id }
+			self.response.out.write(json.dumps(res))
```
Comment (gleono) [inline webapp/services/update_employee.py:32]: Same here for status codes.

File: webapp/services/update_user.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+		if user != None:
+			user.name = user_name
+			user.email = user_email
+			user.date_of_birth = user_date_of_birth
+
+			user.put()
+
+			res = { "msg" : "User successfully updated", "success": True, "user_id" : user_id }
+			self.response.out.write(json.dumps(res))
```
Comment (gleono) [inline webapp/services/update_user.py:30]: and here

File: webapp/www/layout.html [hunk 8]
```diff
@@ -27,7 +29,7 @@
     <div class="collapse navbar-collapse" id="navbarResponsive">
       <ul class="navbar-nav navbar-sidenav" id="exampleAccordion">
         <li class="nav-item" data-toggle="tooltip" data-placement="right" title="Tap List">
-          <a class="nav-link" href="/tap-list">
+          <a class="nav-link" href="/tap_list">
```
Comment (gleono) [inline webapp/www/layout.html:32]: Thank you!

**Review by gleono — CHANGES_REQUESTED** [review]
Please address the key properties in the OrderBeer model and the status code for post methods in the services. Other than that looks great 👍. Good work.

File: webapp/services/update_user.py (hunk 7 above)
Comment (rayolanderos) [reply webapp/services/update_user.py:30]: Using all of these responses as output messages in the layout for error / success alerts. Is there a reason why you want me to update all of this? I am aware of REST protocol but even for ajax a post response in json format is super handy.

File: webapp/controllers/employees.py (hunk 1 above)
Comment (rayolanderos) [reply webapp/controllers/employees.py]: no

**Comment by rayolanderos** [comment]
Ended up leaving keys and post request responses for error msging. 

Removed all useless logging and updated the beer description property to allow for longer text.

**Pull request merged**